api-2.2/src/main/java/org/openmrs/module/fhir/helper/ConditionHelperImpl2_2.java

```diff
@@ -68,6 +67,42 @@ public Condition createCondition(Condition condition) {
 		return generateFHIRCondition(openMrsCondition);
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#updateFHIRCondition(org.hl7.fhir.dstu3.model.Condition)
+	 */
+	@Override
+	public Condition updateFHIRCondition(Condition condition) {
+		if (condition.getId() != null) {
+			ConditionService conditionService = Context.getService(ConditionService.class);
+			org.openmrs.Condition retrievedCondition = conditionService.getConditionByUuid(condition.getId());
+			return retrievedCondition != null ? updateRetrievedCondition(condition, retrievedCondition) :
+					createCondition(condition);
+		} else {
+			throw new UnprocessableEntityException("condition Id cannot be null");
+		}
+
+	}
+
+	private Condition updateRetrievedCondition(Condition condition, org.openmrs.Condition retrievedCondition) {
```

Comment:
What does retried condition mean?

Reply:
is it retried or updateRetrievedCondition? 
don't see retried condition?

Reply:
Sorry i mean what does retrieved condition mean?

Reply:
The condition retrieved from DB
`org.openmrs.Condition retrievedCondition = conditionService.getConditionByUuid(condition.getId());`. If not equals to null then update the retrieved condition

Reply:
So this method is aware of what happens to the parameter before it is passed to it?

Reply:
yeah

Reply:
Would that be a good practice?

Reply:
what do you suggest?

Reply:
Document the method and i will give you my suggestion depending on your intentions for this method.

Reply:
Done

Reply:
@dkayiwa what next?